src/Well.sol

```diff
@@ -423,6 +424,50 @@ contract Well is ERC20PermitUpgradeable, IWell, ReentrancyGuardUpgradeable, Clon
         }
     }
 
+    //////////////////// SHIFT ////////////////////
+
+    // TODO questions: 
+    // do we need to check whether the token is in the pool?
```

Comment:
I don't think so, I think for this it's okay on putting the onus of knowing the tokens in the pool on the dev/user, saves the overhead of having to check the token, even though if they put a token not in the pool an early exit would save them some gas